proposals/dictionary-expressions.md

```diff
@@ -130,41 +186,46 @@ struct Pair<X, Y>
 
 ```c#
-Dictionary<string, Option> optionMap = [opt1Name: opt1Default, opt2Name: opt2Default, .. userProvidedOptions];
+record struct Pair<X, Y>(X x, Y y);
+
+Dictionary<int, string> map1 = [pair1, pair2]; // ?
 ```
 
-Or, conversely:
+Resolution: TBD.  Working group recommendation: Allow anything deconstructible into two values (which includes `KeyValuePair`s as well as user types).  It is reasonable to have domain specific types that represent pairs, and to want to then make maps from those.  Making this pleasant, without requiring contortions to do the mapping explicitly would enable powerful and succinct code. 
 
-```c#
-Dictionary<string, Option> optionMap = [.. Defaults.CoreOptions, feature1Name: feature1Override];
-```
+### Open question 2
 
-Which approach should we go with with our dictionary expressions? Options include:
+We will support *dictionary elements* for C#12 collection expression target types. Which approach should we use for initialization for those types?
 
-1. Purely restrictive.  All elements use `.Add` to be added to the list.  Note: types like `ConcurrentDictionary` would then not work, not without adding support with something like the `CollectionBuilderAttribute`.
-2. Purely permissive.  All elements are added using the indexer.  Perhaps with compiler warnings if the exact same key is given the same constant value twice.
-3. Perhaps a hybrid model.  `.Add` if only using `k:v` and switching to indexers if using spread elements.  There is deep potential for confusion here.
+For example, consider a type like so:
 
-**Resolution:** Use *indexer* as the lowering form. [LDM-2024-03-11](https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-03-11.md#conclusions)
+```c#
+public class Hybrid<TKey, TValue> : IEnumerable<KeyValuePair<TKey, TValue>>
+{
+    public void Add(KeyValuePair<TKey, TValue> pair);
+    public TValue this[TKey key] { ... }
+}
 
-### Open question 6
+// This would compile in C# 12:
+// Translating to calls to .Add.
+Hybrid<string, int> nameToAge = [someKvp];
+```
 
-From [Open Question 3](#open-question-3), we will support *dictionary elements* for C#12 collection expression target types. Which approach should we use for initialization for those types? Options include:
+Options include:
 
 1. Use applicable instance indexer if available; otherwise use C#12 initialization.
 2. Use applicable instance indexer if available; otherwise report an error during construction (or conversion?).
 3. Use C#12 initialization always.
 
+Resolution TBD.  Working group recommendation: Use applicable instance indexer only.  This ensures that everything dictionary-like is initialized in a consistent fashion.  This would be a break in behavior when recompiling.  The view is that these types would be rare.  And if they exist, it would be nonsensical for them to behave differently using the indexer versus the .Add (outside of potentially throwing behavior).
```

Comment:
i again took the strong opinoin that if you have an indexer, you are a dictionary, with dictionary semantics and we should use that.  Existing .Adds were likely to bridge to that anyways.  Feel free to disagree with me here

Reply:
I'm okay with the silent switch from "throw on duplicate" to "overwrite on duplicate."